pkg/api/v1/client/interface.go

```diff
@@ -29,6 +29,9 @@ type Client interface {
 	GetExecutor(name string) (executor testkube.ExecutorDetails, err error)
 	ListExecutors() (executors testkube.ExecutorsDetails, err error)
 	DeleteExecutor(name string) (err error)
+
+	GetExecutionArtifacts(executionID string) (artifacts testkube.Artifacts, err error)
+	DownloadFile(executionID, fileName, destination string) (artifact string, err error)
```

Comment:
io.Reader instead of string should be used

Reply:
(For return values )

Reply:
Dowload file already downloads the file and stores it on the client machine i thought no need to pass the io.Reader

Reply:
The function returns only file name and in the   fuction i construct the file out of datastream

Reply:
Take another look please